pages/docs/manual/latest/bind-to-ts-other.mdx

```diff
@@ -0,0 +1,86 @@
+---
+title: "More TypeScript interop"
+description: "More TypeScript interop"
+canonical: "/docs/manual/latest/bind-to-ts-other"
+---
+
+# More TypeScript interop
+
+## Imported Types
+
+It's possible to import an existing TS type as an opaque type in ReScript. For example,
+
+```res
+@genType.import("./SomeFlowTypes") type weekday
+```
```

Comment:
This is a feature that should be removed in the long term (See https://github.com/rescript-lang/rescript-compiler/issues/6196)

Perhaps we'll need to mark this as deprecated now to get there in v12.

Reply:
Maybe then we don't even cover it anymore in the new docs?
cc @ryyppy

Reply:
IMO, unless we prepare a separate section for deprecated features, yes.

It can be useful sometimes. It's an awkward experience for a user learning genType to discover a feature in another codebase that doesn't exist in the docs. (often happened in bucklescript era)

Reply:
> This is a feature that should be removed in the long term (See [rescript-lang/rescript-compiler#6196](https://github.com/rescript-lang/rescript-compiler/issues/6196))
> 
> Perhaps we'll need to mark this as deprecated now to get there in v12.

It cannot removed yet until the new proposal is implemented right?

Reply:
I think so. I haven't used that feature at all, but I don't know what it's like elsewhere.

Reply:
So I guess it can't be deprecated yet while there's no replacement.

Reply:
But isn't it the same even if the RFC is implemented in v11.x? There is currently no "replacement" for it in the RFC.